sematic/api_client.py

```diff
@@ -115,6 +116,29 @@ def get_graph(run_id: str) -> Tuple[List[Run], List[Artifact], List[Edge]]:
     return runs, artifacts, edges
 
 
+def save_resolution(resolution: Resolution):
+    payload = {
+        "resolution": resolution.to_json_encodable(),
+    }
+    _put(f"/resolutions/{resolution.root_id}", payload)
+
+
+def get_resolution(root_id: str) -> Resolution:
+    """
+    Get resolution
+    """
+    response = _get("/resolutions/{}".format(root_id))
+
+    return Resolution.from_json_encodable(response["content"])
+
+
+def resolution_exists(root_id: str) -> bool:
+    """Determine whether a resolution with the given id exists"""
+    response = _get("/resolutions/{}/exists".format(root_id))
```

Comment:
IIUC this is the only place where we would need to `try/except` around a standard `GET /resolusions/<id>` call. Is it so bad?
Your argument about logs still stands, but I'm not sure it warrants having a separate endpoint.